crates/apub/src/activities/deletion/delete.rs

```diff
@@ -135,25 +108,140 @@ impl ActivityHandler for DeletePostCommentOrCommunity {
   }
 }
 
-async fn get_post_or_comment_actor_id(
+impl Delete {
+  pub(in crate::activities::deletion) async fn send(
+    actor: &Person,
+    community: &Community,
+    object_id: Url,
+    summary: Option<String>,
+    context: &LemmyContext,
+  ) -> Result<(), LemmyError> {
+    let id = generate_activity_id(DeleteType::Delete)?;
+    let delete = Delete {
+      to: PublicUrl::Public,
+      object: object_id,
+      cc: [community.actor_id()],
+      kind: DeleteType::Delete,
+      summary,
+      common: ActivityCommonFields {
+        context: lemmy_context(),
+        id: id.clone(),
+        actor: actor.actor_id(),
+        unparsed: Default::default(),
+      },
+    };
+
+    let activity = AnnouncableActivities::Delete(delete);
+    send_to_community_new(activity, &id, actor, community, vec![], context).await
+  }
+
+  // TODO: would be nice if we could merge this and receive_delete_action() into a single method
+  //       (or merge with receive_undo_delete_action() instead)
+  async fn receive_delete_action(
```

Comment:
There is a lot of duplicate code between receive_delete_action(), receive_remove_action(), receive_undo_delete_action() and receive_undo_remove_action(). I would like to simplify and deduplicate this, by moving the following code into a generic method, so it works for post, comment, community.
```

        let deleted_post = blocking(context.pool(), move |conn| {
          Post::update_deleted(conn, post.id, true)
        })
        .await??;
        send_post_message(deleted_post.id, DeletePost, context).await
```
Problem is that it would need a lot of generic parameters (Post, PostView, PostResponse, SendPost). It would be nice if we could determine them simply based on the Post type. Actually now I'm thinking to make a method Post::send_websocket_message, maybe that would work.

Reply:
I'm guessing this might be a case where the generic version looks uglier and is less clean than the verbose one.

Reply:
Do you want to bet? ;)

Reply:
haha go for it.